subsys/zigbee/lib/zigbee_app_utils/zigbee_app_utils.c

```diff
@@ -856,3 +868,47 @@ void zigbee_configure_sleepy_behavior(bool enable)
 	}
 }
 #endif
+
+/* Wrapper allowing to schedule zb_bdb_reset_via_local_action as an alarm */
+static void factory_reset(zb_uint8_t arg)
+{
+	ZVUNUSED(arg);
+	LOG_DBG("Perform factory reset");
+	zb_bdb_reset_via_local_action(0);
+}
+
+static void factory_reset_callback(struct k_timer *timer_id)
+{
+	uint32_t button_state = 0;
+	uint32_t has_changed = 0;
+
+	dk_read_buttons(&button_state, &has_changed);
+	if (button_state & factory_reset_context.button) {
+		LOG_DBG("FR button pressed for %d [s]", timer_id->status);
+		if (timer_id->status >= factory_reset_context.press_time) {
+			ZB_SCHEDULE_APP_ALARM(factory_reset, 0, FACTORY_RESET_DELAY);
```

Comment:
Why do we need a `FACTORY_RESET_DELAY` delay here?

Reply:
@tomchy When factory_reset function was called here directly I observed crashes. Expecting some scheduling/timing problems, I tried to call it after scheduling an alarm - this approach helped and that is why it is visible. Now I tried to do the scheduling with 0 instead of 100 ms (FACTORY_RESET_DELAY's value) and did not observe crashes on Thingy53 ZWS app but cannot tell about other samples. I would like to discuss this aspect but have difficulties reaching you.